client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java

```diff
@@ -249,10 +302,126 @@ public ClientSession toClientSession()
                 .build();
     }
 
+    public TrinoUri getTrinoUri()
+    {
+        return getTrinoUri(Collections.emptyList());
+    }
+
+    public TrinoUri getTrinoUri(List<String> providedOptions)
```

Comment:
can we extract this method to a separate class and provide some tests for it?

Reply:
There are lots of test for it in `TestClientOptions` already.